Initial import of toolchain image sanity tests

happz opened the pull request:
Set of very simple tests, to verify basic usability of tools packaged in the image.

happz committed:
Initial import of toolchain image sanity tests

Set of very simple tests, to verify basic usability of tools packaged in
the image.

hhorak commented on 4-toolchain/test/run:
```diff
@@ -0,0 +1,153 @@
+function test_sanity_gcc_usage () {
+  info "Testing 'gcc -v' usage ..."
+
+  docker run --rm $IMAGE_NAME gcc -v &> $TMPDIR/actual-gcc-v
```
Would it be possible to compare only stdout? In my case the daemon returns some warning to stderr, because I don't have my docker daemon configured properly and that causes the test to fail. I believe checking the stdout only should be good enough test.

happz replied:
Fair enough. Commit 36d9feb changes this.

hhorak commented:
Except the one comment about checking the usage, I don't have anything else.

happz committed:
In toolchain usage test, capture and compare only stdout

Avoid capturing stderr to avoid spoiling "docker run ..." output with
accidental warnings from Docker tools.

hhorak commented:
Thank you, LGTM.
